glean-core/ios/GleanTests/Scheduler/MetricsPingSchedulerTests.swift

```diff
@@ -0,0 +1,334 @@
+        // the previous run) but must not send the canary string, which would be sent the next time
+        // the "metrics" ping is collected after this one.
+        // Glean.shared.initialize(uploadEnabled: true)
+        Glean.shared.initialize(uploadEnabled: true)
+        wait(for: [expectation!], timeout: TimeInterval(5.0))
+
+        // Clean up
+        Glean.shared.resetGlean(clearStores: true)
+        Glean.shared.testDestroyGleanHandle()
+    }
+
+    func testGleanPreservesLifetimeApplicationMetrics() {
+        // Reset Glean and do not start it right away
+        clearPingDirectory()
+        Glean.shared.testDestroyGleanHandle()
+        Dispatchers.shared.setTaskQueuing(enabled: true)
+
+        // Set the last time the "metrics" ping was set to now. This is required for us to not
+        // send a metrics pings the first time we initialize Glean.
+        let now = Date()
+        Glean.shared.metricsPingScheduler.updateSentDate(now)
+
+        // Create a metric and set its value. We expect this to be sent in the first ping
+        // that gets generated the SECOND time we start Glean.
+        let testMetric = StringMetricType(
+            category: "telemetry",
+            name: "test_applifetime_metric",
+            sendInPings: ["metrics"],
+            lifetime: Lifetime.application,
+            disabled: false
+        )
+        let expectedValue = "I-will-survive!"
+
+        // Reset Glean and start it for the FIRST time, then record a value.
+        Glean.shared.resetGlean(clearStores: false)
+        testMetric.set(expectedValue)
+
+        // Set the last time the "metrics" ping was sent to yesterday, which should make
+        // the ping overdue and trigger it at startup.
+        let yesterday = Calendar.current.date(byAdding: Calendar.Component.day, value: -1, to: now)
+        Glean.shared.metricsPingScheduler.updateSentDate(yesterday!)
+
+        // Set up the interception of the ping for inspection
+        let host = URL(string: Configuration.Constants.defaultTelemetryEndpoint)!.host!
+        stub(condition: isHost(host)) { data in
+            let body = (data as NSURLRequest).ohhttpStubs_HTTPBody()
+            let json = try! JSONSerialization.jsonObject(with: body!, options: []) as? [String: Any]
+            XCTAssert(json != nil)
+            let metrics = json?["metrics"] as? [String: Any]
+            let strings = metrics?["string"] as? [String: Any]
+
+            // Ensure there is only the expected metric
+            XCTAssertEqual(1, strings?.count, "Must contain only the expected metric")
+
+            // Check the received metric's value against the expected value
+            let receivedValue = strings?["telemetry.test_applifetime_metric"] as? String
+            XCTAssertEqual(expectedValue, receivedValue, "Values must match")
+
+            DispatchQueue.main.async {
+                // let the response get processed before we mark the expectation fulfilled
```

Comment:
I wonder if we should do this for our other tests as well 🤔 right now we fulfill, then return an HTTP status.

Reply:
Happy to file a bug if you think it's worth it.

Reply:
Yup, definitel worth filing. It's an easy change.

Reply:
[Bug filed](https://bugzilla.mozilla.org/show_bug.cgi?id=1610516)